pkg/vault/error.go

```diff
@@ -11,22 +11,39 @@ import (
 	"syscall"
 )
 
-// Error is the custom error type for this package
-type Error struct {
+// CustomVaultError is the custom error type for this package
+type CustomVaultError struct {
 	MessageParts  []string
 	OriginalError error
 }
 
+// Error is custom error interface with one method that will block other functions
+// from using this Error. This is not interchangeable with the standard error.
+type Error interface {
```

Comment:
Personally I would change this error name to, could possibly get away with keeping `CustomVaultError` private `customVaultError` and then only using the interface everywhere